File: code-examples/javascript/Mod3/3.01/test/LocatorTest.js
const { Builder, By } = require('selenium-webdriver')
const path = require('path')

//The Test
describe('Locate', function() {
   this.timeout(30000)
   let driver

   beforeEach(async function() {
   driver = await new Builder().forBrowser('firefox').build()
   })
   afterEach(async function() {
       await driver.quit()
   })
   it('check red button text' , async function() {
    await driver.get('https://the-internet.herokuapp.com/challenging_dom')
    await driver
        .findElement(By.className('button success'))
        .click()
    await driver
        .findElement(By.className('button'))
        .click()
        // Return the text of the red button id=button alert contains 'foo', 'bar', 'baz', or 'qux'
    var redButtonMessage = await driver
        .findElement(By.className('button alert'))
        .getText()
    console.log(redButtonMessage)
    }) 
})
   

File: python/Mod2/2.07/tests/locator_test.py
# filename: tests/locator_test.py
import pytest
import os
from selenium import webdriver
from selenium.webdriver.common.by import By

@pytest.fixture
def driver(request):
    _chromedriver= os.path.join(os.getcwd(), 'vendor', 'chromedriver')
    if os.path.isfile(_chromedriver):
        driver_ = webdriver.Chrome(_chromedriver)

    else:
        driver_ = webdriver.Chrome()

    def quit():
        driver_.quit()

    request.addfinalizer(quit)
    return driver_

def test_valid_credentials(driver):
    driver.get("https://the-internet.herokuapp.com/challenging_dom")
    driver.find_element(       , "   ").click()
    driver.find_element(    , "   ").click()
    printredbutton = driver.find_element(       , "   ").text
    assert (printredbutton == 'foo'), print(printredbutton)
